src/ol/render/canvas/TextBuilder.js

```diff
@@ -240,7 +239,6 @@ class CanvasTextBuilder extends CanvasBuilder {
         case GeometryType.POINT:
         case GeometryType.MULTI_POINT:
           flatCoordinates = /** @type {import("../../geom/MultiPoint.js").default} */ (geometry).getFlatCoordinates();
-          end = flatCoordinates.length;
```

Comment:
I hadn’t yet approved because I was curious to know why we still need to set `end` for MultiLineString and why we still set `stride` for Polygon.  But I haven’t yet made the time to understand it.

Reply:
Thanks for the review @tschaub!

For polygons, the `getFlatMidpoint()` function always uses a stride of 3, with the available length at the midpoint at index 2. That is why we set the `stride` to 3.

For the `getFlatMidpoints` result in the MultiLineString, I'm going to add another test case. Looks like the fact that we still set `ends` there might be causing issues.

Reply:
Thanks for raising concerns @tschaub. There were indeed more issues that I fixed now. I also added tests for all geometry types.

Reply:
We need to set `stride = 2` for MultiLineString because `getFlatMidpoints()` returns 2-dimensional coordinates. And for MultiPolygon, because we create a 2-dimensional `flatCoordinates` array ourselves. For all other geometry types, we only receive a single point in `flatCoordinates`, of which only the first two values are used anyway because `end = 2` is set.